mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1175,6 +1354,47 @@ public void close() {
 		}
 	}
 
+	private static class SingleTupleIterator implements ClosableIterator<Tuple> {
+		private DBObject theOne;
+		private final EntityKeyMetadata metadata;
+		private final DBCollection collection;
+
+		public SingleTupleIterator(DBObject theOne, DBCollection collection, EntityKeyMetadata metadata) {
+			this.theOne = theOne;
+			this.collection = collection;
+			this.metadata = metadata;
+		}
+
+		@Override
+		public boolean hasNext() {
+			return theOne != null;
+		}
+
+		@Override
+		public Tuple next() {
+			if (theOne == null)
```

Comment:
No. Remind that next() or remove() may be invoked erroneously more than once, and that should fail for all but the first invocation.